quartz/components/Head.tsx

```diff
@@ -1,30 +1,172 @@
 import { FullSlug, _stripSlashes, joinSegments, pathToRoot } from "../util/path"
 import { JSResourceToScriptElement } from "../util/resources"
 import { QuartzComponentConstructor, QuartzComponentProps } from "./types"
+import satori, { SatoriOptions } from "satori"
+import * as fs from "fs"
+import { ImageOptions, SocialImageOptions, getSatoriFont } from "../util/imageHelper"
+import sharp from "sharp"
+import { defaultImage } from "../util/socialImage"
+import { unescapeHTML } from "../util/escape"
+
+/**
+ * Generates social image (OG/twitter standard) and saves it as `.webp` inside the public folder
+ * @param opts options for generating image
+ */
+async function generateSocialImage(opts: ImageOptions, userOpts: SocialImageOptions) {
+  const { cfg, description, fileName, fontsPromise, title } = opts
+  const fonts = await fontsPromise
+
+  // JSX that will be used to generate satori svg
+  const imageElement = userOpts.imageStructure(cfg, userOpts, title, description, fonts)
+
+  const svg = await satori(imageElement, {
+    width: userOpts.width,
+    height: userOpts.height,
+    fonts: fonts,
+  })
+
+  // Convert svg directly to webp (with additional compression)
+  const compressed = await sharp(Buffer.from(svg)).webp({ quality: 40 }).toBuffer()
+
+  // Write to file system
+  fs.writeFileSync(`${imageDir}/${fileName}.${extension}`, compressed)
+}
+
+const extension = "webp"
+const imageDir = "public/static/social-images"
+
+const defaultOptions: SocialImageOptions = {
+  colorScheme: "lightMode",
+  width: 1200,
+  height: 676,
```

Comment:
```suggestion
  height: 630,
```

internet seems to suggest 630px tall